src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentGeneratedOutputTracker.cs

```diff
@@ -111,7 +111,14 @@ private async Task<RazorCodeDocument> GetGeneratedOutputInitializationTaskCore(P
 
             var projectEngine = project.GetProjectEngine();
 
-            return projectEngine.ProcessDesignTime(documentSource, importSources, tagHelpers);
+            var codeDocument = projectEngine.ProcessDesignTime(documentSource, importSources, tagHelpers);
+            var csharpDocument = codeDocument.GetCSharpDocument();
+            if (document is DefaultDocumentSnapshot defaultDocument)
```

Comment:
You should just pass the `DefaultDocumentSnapshot` as a parameter.

Reply:
Hmm, sure 👍

Reply:
Actually, can't do that. [DocumentSnapshots are used for import tracking](https://github.com/aspnet/Razor/blob/96709c4d7723d20f8604e2313606708d44233cc7/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentImportsTracker.cs#L87) as well and they need to be able to generate content.  It'd be a re-design of that feature to make this method able to take a `DefaultDocumentSnapshot`. Having this here enables non-default documents to generate output and if they happen to be a default one we set their corresponding `HostDocuments` output.

Reply:
ugh